sql/moz-fx-data-shared-prod/telemetry_derived/newtab_interactions_v2/query.sql

```diff
@@ -0,0 +1,291 @@
+    ) as had_non_impression_engagement
+    FROM
+      events_unnested
+  GROUP BY
+    newtab_visit_id,
+    submission_date
+  HAVING
+    newtab_visit_id IS NOT NULL
+),
+search_events AS (
+  SELECT
+    mozfun.map.get_key(event_details, "newtab_visit_id") AS newtab_visit_id,
+    mozfun.map.get_key(event_details, "telemetry_id") AS search_engine,
+    mozfun.map.get_key(event_details, "search_access_point") AS search_access_point,
+    COUNTIF(event_name = "issued") AS searches,
+    COUNTIF(
+      event_name = "impression"
+      AND mozfun.map.get_key(event_details, "is_tagged") = "true"
+    ) AS tagged_search_ad_impressions,
+    COUNTIF(
+      event_name = "impression"
+      AND mozfun.map.get_key(event_details, "is_follow_on") = "true"
+    ) AS follow_on_search_ad_impressions,
+    COUNTIF(
+      event_name = "impression"
+      AND mozfun.map.get_key(event_details, "is_follow_on") = "true"
+      AND mozfun.map.get_key(event_details, "is_tagged") = "true"
+    ) AS tagged_follow_on_search_ad_impressions,
+    COUNTIF(
+      event_name = "click"
+      AND mozfun.map.get_key(event_details, "is_tagged") = "true"
+    ) AS tagged_search_ad_clicks,
+    COUNTIF(
+      event_name = "click"
+      AND mozfun.map.get_key(event_details, "is_follow_on") = "true"
+    ) AS follow_on_search_ad_clicks,
+    COUNTIF(
+      event_name = "click"
+      AND mozfun.map.get_key(event_details, "is_follow_on") = "true"
+      AND mozfun.map.get_key(event_details, "is_tagged") = "true"
+    ) AS tagged_follow_on_search_ad_clicks
+  FROM
+    events_unnested
+  WHERE
+    event_category IN ('newtab.search', 'newtab.search.ad')
+  GROUP BY
+    newtab_visit_id,
+    search_engine,
+    search_access_point
+),
+search_summary AS (
+  SELECT
+    newtab_visit_id,
+    ARRAY_AGG(
+      STRUCT(
+        search_engine,
+        search_access_point,
+        searches,
+        tagged_search_ad_clicks,
+        tagged_search_ad_impressions,
+        follow_on_search_ad_clicks,
+        follow_on_search_ad_impressions,
+        tagged_follow_on_search_ad_clicks,
+        tagged_follow_on_search_ad_impressions
+      )
+    ) AS search_interactions
+  FROM
+    search_events
+  GROUP BY
+    newtab_visit_id
+),
+topsites_events AS (
+  SELECT
+    mozfun.map.get_key(event_details, "newtab_visit_id") AS newtab_visit_id,
+    SAFE_CAST(mozfun.map.get_key(event_details, "position") AS INT64) AS topsite_position,
+    mozfun.map.get_key(event_details, "advertiser_name") AS topsite_advertiser_name,
+    mozfun.map.get_key(event_details, "tile_id") AS topsite_tile_id,
+    COUNTIF(event_name = 'impression') AS topsite_impressions,
+    COUNTIF(event_name = 'click') AS topsite_clicks,
+    COUNTIF(
+      event_name = 'impression'
+      AND mozfun.map.get_key(event_details, "is_sponsored") = "true"
+    ) AS sponsored_topsite_impressions,
+    COUNTIF(
+      event_name = 'impression'
+      AND mozfun.map.get_key(event_details, "is_sponsored") = "false"
```

Comment:
I might just be careful here, with this formulation it _could_ end up in a state where `sponsored_topsite_impressions + organic_topsite_impressions != topsite_impressions`.

If you instead set `organic_topsite_impressions = sponsored_topsite_impressions - topsite_impressions`, it would guarantee they are the same (this might not be possible, in which case, ignore)

Reply:
Same for clicks

Reply:
I think we'd want to catch if there is a discrepancy like that, though I can't think of a case where the is_sponsored is neither true nor false so I would expect it to be a very rare thing.  I'll defer to your judgment on whether to handle this here on in Looker.